### PR TITLE
Frame Pass API [2019.2]

### PR DESCRIPTION
**Pull request by fredpointzero**
### Purpose of this PR
Provide an API to manipulate rendered data of a frame with specific settings.

---
### Release Notes
Added an API to render specific settings during a rendering.

---
### Testing status
**Katana Tests**: [ABV Running](https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=master&ScriptableRenderLoop_branch=HDRP%2FframepassApi&unity_branch=trunk)

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: -

---
### Overall Product Risks
**Technical Risk**: Medium

**Halo Effect**: Low

---
### Comments to reviewers
Notes for the reviewers you have assigned.

---

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassForward.hlsl [hunk 1]
```diff
@@ -160,61 +160,60 @@ void Frag(PackedVaryingsToPS packedInput,
     if (!viewMaterial)
     {
         if (_DebugFullScreenMode == FULLSCREENDEBUGMODE_VALIDATE_DIFFUSE_COLOR || _DebugFullScreenMode == FULLSCREENDEBUGMODE_VALIDATE_SPECULAR_COLOR)
-        {
-            float3 result = float3(0.0, 0.0, 0.0);
+    {
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassForward.hlsl]: Why is there space here ? shouldn't be required

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassForwardUnlit.hlsl [hunk 2]
```diff
@@ -65,24 +65,24 @@ float4 Frag(PackedVaryingsToPS packedInput) : SV_Target
     {
         int indexMaterialProperty = int(_DebugViewMaterialArray[index]);
         if (indexMaterialProperty != 0)
-        {
-            float3 result = float3(1.0, 0.0, 1.0);
-            bool needLinearToSRGB = false;
+    {
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassForwardUnlit.hlsl]: why is there a shift of space here, don't do it

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs [hunk 3]
```diff
@@ -214,6 +215,11 @@ public HDAdditionalCameraData.FlipYMode flipYMode
 
         public HDPhysicalCamera physicalParameters => m_AdditionalCameraData?.physicalParameters;
 
+        public IEnumerable<AOVRequestData> framePasses =>
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Camera/HDCamera.cs]: still framePasses?

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 4]
```diff
@@ -1405,16 +1405,38 @@ ref _cullingResults
                             target.id = m_TemporaryTargetForCubemaps;
                         }
 
+
+                        var framePassIndex = 0;
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: framePass here too

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/AOV/AOVRequestBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace UnityEngine.Experimental.Rendering.HDPipeline
+{
+    /// <summary>Utility to build frame passes.</summary>
+    public class AOVRequestBuilder : IDisposable
+    {
+        // Owned
+        private List<AOVRequestData> m_AOVRequestDataData;
+
+        /// <summary>Add a frame pass.</summary>
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/AOV/AOVRequestBuilder.cs]: Frame pass

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/AOV/AOVRequestData.cs [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+using UnityEngine.Rendering;
+
+namespace UnityEngine.Experimental.Rendering.HDPipeline
+{
+    /// <summary>Called when the rendering has completed.</summary>
+    /// <param name="cmd">A command buffer that can be used.</param>
+    /// <param name="buffers">The buffers that has been requested.</param>
+    /// <param name="outputProperties">Several properties that were computed for this frame.</param>
+    public delegate void FramePassCallback(CommandBuffer cmd, List<RTHandleSystem.RTHandle> buffers, RenderOutputProperties outputProperties);
+    public delegate RTHandleSystem.RTHandle FramePassBufferAllocator(AOVBuffers aovBufferId);
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/AOV/AOVRequestData.cs]: FramePass

**Comment by sebastienlagarde** [comment]
Katana green (failing test was from master), FrmaePass have been renamed. Merging

**Pull request merged**